test/ruby/test_iseq.rb

```diff
@@ -574,6 +574,19 @@ def test_iseq_builtin_to_a
     assert_equal([:func_ptr, :argc, :index, :name], invokebuiltin[1].keys)
   end
 
+  def test_iseq_builtin_load
+    Tempfile.create(["builtin", ".iseq"]) do |f|
+      f.binmode
+      f.write(RubyVM::InstructionSequence.of(1.method(:abs)).to_binary)
+      f.close
+      assert_normal_exit("#{<<~"begin;"}\n#{<<~'end;'}")
+      path = ""#{f.path.dump}
```

Comment:
The `""#{f.path.dump}` part seems odd to me. It also seems odd this checks for a normal exit, when the use of a builtin function inside an iseq should raise an ArgumentError when loading.   Maybe we want something like this:

```diff
+      assert_raise(ArgumentError) do
+        RubyVM::InstructionSequence.load_from_binary(File.binread(f.path))
+      end
```

Of if we do want to use a separate process, check that it produces an error instead of exits normally?

The compile.c changes do look correct and I tested that the patch does fix the crash.

Reply:
`""#{f.path.dump}` is to tell ruby-mode.el that the rest is not in the string literal.